modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -8744,6 +8749,183 @@ public void testPostRelatedObjectEntryInDifferentCompany()
 		);
 	}
 
+	@FeatureFlags("LPD-37056")
+	@Test
+	public void testPostScopeScopeKeyValidate() throws Exception {
+		ObjectField textObjectField = _objectFieldLocalService.getObjectField(
+			_siteScopedObjectDefinition1.getObjectDefinitionId(),
+			_OBJECT_FIELD_NAME_1);
+
+		String error1 = RandomTestUtil.randomString();
+		String error2 = RandomTestUtil.randomString();
+
+		ObjectValidationRule objectValidationRule =
+			_objectValidationRuleLocalService.addObjectValidationRule(
+				StringUtil.randomString(), TestPropsValues.getUserId(),
+				_siteScopedObjectDefinition1.getObjectDefinitionId(), true,
+				ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+				LocalizedMapUtil.getLocalizedMap(error1),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+				textObjectField.getName() + " == \"foo\"", false,
+				Collections.emptyList());
+
+		_objectValidationRuleLocalService.addObjectValidationRule(
+			StringUtil.randomString(), TestPropsValues.getUserId(),
+			_siteScopedObjectDefinition1.getObjectDefinitionId(), true,
+			ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+			LocalizedMapUtil.getLocalizedMap(error2),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_PARTIAL_VALIDATION,
+			"NOT(isInteger(" + textObjectField.getName() + "))", false,
+			Arrays.asList(
+				new ObjectValidationRuleSettingBuilder(
+				).name(
+					ObjectValidationRuleSettingConstants.
+						NAME_OUTPUT_OBJECT_FIELD_ID
+				).value(
+					String.valueOf(textObjectField.getObjectFieldId())
+				).build()));
+
+		User user1 = _addUser("test1", "test1");
+
+		_resourcePermissionLocalService.removeResourcePermission(
+			TestPropsValues.getCompanyId(), ObjectEntry.class.getName(),
+			ResourceConstants.SCOPE_INDIVIDUAL,
+			String.valueOf(_objectEntry5.getObjectEntryId()),
+			user1.getRoleIds()[0], ActionKeys.ADD_ENTRY);
+
+		_setUpPermissionThreadLocal(user1);
```

Comment:
When changing ThreadLocal values, you should return the original values ​​in a try finally statement so as not to interfere with other tests.
Please see [this](https://github.com/liferay/liferay-portal/blob/15191a5ee86370e85705fa470f526953c124f4e9/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java#L6798-L6804) example in the `testGetObjectEntryWithAuditEvents` method.